### ISSUE TITLE
Having trouble reaching the performance in the paper for the first layer

### ISSUE THREAD

**Issue by jlindsey15** [eugenium/layerCNN#1]
Hi!  When I run the k=1 ImageNet code I get only 3-4% top-5 accuracy after training the first layer for the full 45 epochs.  Figure 2 of the paper suggests I should be getting over 20%.  Any ideas why this might be happening?

EDIT: I realize it could be the case that the first layer is meant to continue training even after the other layers are introduced.  That could account for what I am observing, but I want to make sure

**Comment by eugenium** [comment]
Hi, thanks for the interest. So no the model should not continue to be trained after (not in this code at least). There shouldn't be much variability in the final top-5 either. 

My best guess is that the problem is related to versions of pytorch as there has been many changes in pytorch the last 1-2 years that change behavior of underlying functions without warning. I suggest dropping down to pytorch 0.4 or 0.3. I will try to check the exact versions of pytorch/torchvision we used and get back to you.

**Comment by eugenium** [comment]
I checked, you should try pytorch 0.4 and let me know if you still see the problem. You should be able to find the issue without rerunning the full training because looking at my logs the top5 accuracy after a single epoch on the first layer is around 12% (and 7% running avg train). 

n: 0, epoch 0, train top1:2.379237054962012(top5:7.451487589063442), test top1:4.126 (top5:12.226)

If you need the full log files to help you debug send me an email.

**Comment by jlindsey15** [comment]
Thanks for the response -- unfortunately, switching to PyTorch 0.4 didn't help the issue.  Any other advice you have would be appreciated.

**Comment by eugenium** [comment]
I would need more information. Are you using the standalone file imagenet_single_layer.py, whats the command run?

**Comment by jlindsey15** [comment]
Yes, my command run is:

python imagenet_single_layer.py [IMAGENET_DIRECTORY] --ncnn 8 --save_folder ./published_model

Logs for the first two epochs are as follows:

n: 0, epoch 0, train top1:0.31510332226753235(top5:1.5057365894317627), test top1:0.44200000166893005 (top5:1.937999963760376), top1ens:0.44200003147125244 top5ens:1.9380000829696655
n: 0, epoch 1, train top1:0.435150146484375(top5:2.0098862648010254), test top1:0.44999998807907104 (top5:2.385999917984009), top1ens:0.45000001788139343 top5ens:2.386000156402588

**Comment by eugenium** [comment]
Just to be sure have you used the same imagenet folder before to train imagenet with a standard pytorch pipeline (e.g. https://github.com/pytorch/examples/blob/master/imagenet/main.py) ? There can be issues sometimes with the directory structure (e.g. hidden files), ImageFolder dataset constructor doesnt really have any checks to make sure the dataset is in the good format.

**Comment by jlindsey15** [comment]
Ah yep, the problem is indeed with the ImageNet directory structure.  Sorry for the trouble, this was totally my bad!  Thanks for your time and for this neat repo :)

(A description of the problem in the unlikely event that it helps someone some day: My class labels are offset by one when I load them into pytorch.  I am used to subtracting one from them to account for this, but here I only did so for "target_var" and not for "target," meaning the model was training on one set of labels but the accuracy results being printed were for another offset set of labels.)
